### ISSUE TITLE
Possible negative loss value in bimodal_loss

### ISSUE THREAD

**Issue by JagerU** [fabiotosi92/SMD-Nets#5]
The idea of bimodal mixture density is brilliant! 
However I notice that, theoretically, the loss function can produce negative values in certain circumstances. And I actually got negative losses during the training procedure.

In SMD-Nets/lib/model/losses.py we have:

`def laplacian(mu, b, labels):
    return 0.5 * torch.exp(-(torch.abs(mu-labels)/b))/b
def bimodal_loss(mu0, mu1, sigma0, sigma1, w0, w1, labels, dist="gaussian"):
    return - torch.log(w0 * distribution(mu0, sigma0, labels, dist) + \
                       w1 * distribution(mu1, sigma1, labels, dist))`

Negative values come out for -log(x) when x>1, where x is related to 'exp(-(abs(diff)/b))/b'. We can have, for example, when diff=b=0.01, w0=1, the loss value would be: -log(1* 0.5*exp(-0.01/0.01)/0.01) = -2.91
The function of  'y = 0.5 * exp(x/b)/b'  can be plotted to imply that, for all 'x', only when b<0.2 can achieve y<1

I wonder is it normal to have negative values during your training.
or if there are any more restriction applied on mu, sigma and w0 to prevent this from happening.

**Comment by fabiotosi92** [comment]
Hi! Yes, it is normal to have negative values during training using the negative log likelihood loss function, no worries.  The gradient tells the optimizer how to change the network parameters in order to reduce the overall loss, but it doesn’t care if it is positive or negative.

**Comment by raozhongyu** [comment]
In this way, I'd like to konw is the loss -3.5 smaller than -2.5?
